### DISCUSSION TITLE
Intellisense Warnings for C++ 20 ranges

### DISCUSSION THREAD

**Discussion by coulanuk** [microsoft/vscode-cpptools#8026] [Q&A]
Running the sample code for C++ 20 ranges taken from https://en.cppreference.com/w/cpp/ranges.

VSCode Intellisense warnings appear for :
- no operator "|" matches these operands -- operand types are: std::ranges::filter_view<std::ranges::iota_view<int, int>, type> | std::ranges::views::__adaptor::_RangeAdaptorClosure<type>
- this range-based 'for' statement requires a suitable "begin" function and none was found

The code compiles and runs correctly.
compiler args -std=c++20 are set in tasks.json and c_cpp_properties.json.
See output of C/C++ Log Diagnostics below.
Q: Is there anything else I need to set somewhere to get a clean response from Intellisense?

- OS : Ubuntu 20.04 (Linux x64 5.4.0-81-generic)
- VS Code version : 1.59.0
- Compiler : gcc 10.3.0

-------- Diagnostics - 21/08/2021, 18:27:33
Version: 1.6.0-insiders3
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/gcc",
    "cStandard": "gnu17",
    "cppStandard": "gnu++20",
    "intelliSenseMode": "gcc-x64",
    "compilerArgs": [
        "-std=c++20",
        "-g"
    ],
    "browse": {
        "limitSymbolsToIncludedHeaders": false
    },
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true
}
Translation Unit Mappings:
[ /home/coulan/src2/cpp20/testcpp20.cpp ]:
    /home/coulan/src2/cpp20/testcpp20.cpp
Translation Unit Configurations:
[ /home/coulan/src2/cpp20/testcpp20.cpp ]:
    Process ID: 7926
    Memory Usage: 407 MB
    Compiler Path: /usr/bin/gcc
    Includes:
        /usr/include/c++/10
        /usr/include/x86_64-linux-gnu/c++/10
        /usr/include/c++/10/backward
        /usr/lib/gcc/x86_64-linux-gnu/10/include
        /usr/local/include
        /usr/include/x86_64-linux-gnu
        /usr/include
    Standard Version: c++20
    IntelliSense Mode: linux-gcc-x64
    Other Flags:
        --g++
        --gnu_version=100300
Total Memory Usage: 407 MB

------- Workspace parsing diagnostics -------

**Comment by sean-mcmanus** [comment]
I created a bug at https://github.com/microsoft/vscode-cpptools/issues/8039

**Reply by coulanuk** [reply]
Thanks Sean
